src/libstd/sync/condvar.rs

```diff
@@ -196,12 +188,11 @@ impl Condvar {
     /// }
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
-    pub fn wait<'a, T>(&self, guard: MutexGuard<'a, T>)
+    pub fn wait<'a, T>(&self, mut guard: MutexGuard<'a, T>)
                        -> LockResult<MutexGuard<'a, T>> {
-        let poisoned = unsafe {
-            let lock = mutex::guard_lock(&guard);
-            self.verify(lock);
```

Comment:
Why is the `verify` not needed any more?

Reply:
Because what it did was make sure the same condvar was not being used by two different mutexes. That restriction has now been somewhat lifted, and the remaining limitations are handled internally by `parking_lot::Condvar` for us.